### PR TITLE
Move GeoJSON Types from Azure Core to Azure Core Experimental

### PR DESCRIPTION
**Pull request by alzimmermsft**
This PR moves the GeoJSON types from Azure Core to Azure Core Experimental. In addition to moving these classes Azure Search is updating/removing its use of the GeoJSON types.

### CODE REVIEW

File: sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/spatial/GeometryDeserializer.java [hunk 1]
```diff
@@ -1,18 +1,8 @@
...
-import com.azure.core.models.spatial.Geometry;
-import com.azure.core.models.spatial.GeometryBoundingBox;
-import com.azure.core.models.spatial.GeometryPosition;
-import com.azure.core.models.spatial.LineGeometry;
-import com.azure.core.models.spatial.MultiLineGeometry;
-import com.azure.core.models.spatial.MultiPointGeometry;
-import com.azure.core.models.spatial.MultiPolygonGeometry;
-import com.azure.core.models.spatial.PointGeometry;
-import com.azure.core.models.spatial.PolygonGeometry;
+package com.azure.core.experimental.spatial;
```
Comment (srnagar) [inline sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/spatial/GeometryDeserializer.java:4]: There's a `com.azure.core.experimental.serializer` too. Maybe we should put this in `com.azure.core.experimental.spatial.serializer`?

Comment (alzimmermsft) [reply sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/spatial/GeometryDeserializer.java:4]: Placed the serializers in spatial to denote they are tied to the experimental concept of spatial and not serializers.

**Pull request merged**